File: sender.clj
(ns example.eventbus-pubsub.sender
  (:require [vertx.core :as vertx]
            [vertx.eventbus :as eb]))

(vertx/periodic 1000
                (eb/publish "news-feed" "some news from Clojure")
                (println "published message some news"))

File: Sender.java
import org.vertx.java.core.Handler;
import org.vertx.java.core.eventbus.Message;
import org.vertx.java.platform.Verticle;

public class Sender extends Verticle {

  public void start() {
    vertx.setPeriodic(1000, new Handler<Long>() {
      @Override
      public void handle(Long timerID) {
        vertx.eventBus().publish("news-feed", "Java!");
      }
    });
  }
}
